### DISCUSSION TITLE
What is the management_metrics_collection feature?

### DISCUSSION THREAD

**Discussion by bodograumann** [rabbitmq/rabbitmq-server#10905] [Questions] [answered]
We recently got the following warning:

> 2024-03-13 19:04:00.669077+00:00 [warning] <0.745.0> Deprecated features: `management_metrics_collection`: Feature `management_metrics_collection` is deprecated.
> 2024-03-13 19:04:00.669077+00:00 [warning] <0.745.0> By default, this feature can still be used for now.
> 2024-03-13 19:04:00.669077+00:00 [warning] <0.745.0> Its use will not be permitted by default in a future minor RabbitMQ version and the feature will be removed from a future major RabbitMQ version; actual versions to be determined.
> 2024-03-13 19:04:00.669077+00:00 [warning] <0.745.0> To continue using this feature when it is not permitted by default, set the following parameter in your configuration:
> 2024-03-13 19:04:00.669077+00:00 [warning] <0.745.0>     "deprecated_features.permit.management_metrics_collection = true"
> 2024-03-13 19:04:00.669077+00:00 [warning] <0.745.0> To test RabbitMQ as if the feature was removed, set this in your configuration:
> 2024-03-13 19:04:00.669077+00:00 [warning] <0.745.0>     "deprecated_features.permit.management_metrics_collection = false"

I couldn't find any information about what this `management_metrics_collection` feature is though. Does anybody know?

**Comment by lukebakken** [comment, marked as answer]
https://www.rabbitmq.com/blog/2021/08/21/4.0-deprecation-announcements#disable-metrics-delivery-via-the-management-api--ui

**Comment by Paulitos** [comment]
And how do we get rid off the warning @lukebakken? I can't find a way around to solve it even if we focus on Prometheus.

**Reply by michaelklishin** [reply]
Prometheus users can [disable metric collection in the management UI](https://www.rabbitmq.com/docs/management#disable-stats).

**Reply by michaelklishin** [reply]
IIRC this might have been the feature we have decided to un-deprecate but I cannot immediately say if that's the case in `v4.0.x`. In any case, if you use Prometheus for metrics, disabling stats in the management plugin makes sense, or at least is an option.
